File: AppSets-iOS/AppSets/com/app/appsets/im/ImMessageDesignType.swift
//
//  MQMessageDesignType.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/14.
//

import Foundation

struct ImMessageDesignType {
    public static let TYPE_TEXT = "im.text"
    public static let TYPE_IMAGE = "im.image"
    public static let TYPE_VIDEO = "im.video"
    public static let TYPE_MUSIC = "im.music"
    public static let TYPE_VOICE = "im.voice"
    public static let TYPE_LOCATION = "im.location"
    public static let TYPE_FILE = "im.file"
    public static let TYPE_HTML = "im.html"
    public static let TYPE_AD = "im.ad"
    public static let TYPE_SYSTEM = "im.system"
    public static let TYPE_CUSTOM = "im.custom.*"
    
    public static func getType(_ imMessage: any ImMessage) -> String {
        switch imMessage{
        case is IM_Text:
            return TYPE_TEXT
            
        case is IM_System:
            return TYPE_SYSTEM
        default:
            return TYPE_TEXT
        }
    }
}

File: AppSets-Servers-Spring/MainServer/src/main/kotlin/xcj/app/main/im/ImMessageDesignType.kt
package xcj.app.main.im

object ImMessageDesignType {
    const val TYPE_TEXT = "im.text"
    const val TYPE_IMAGE = "im.image"
    const val TYPE_VIDEO = "im.video"
    const val TYPE_MUSIC = "im.music"
    const val TYPE_VOICE = "im.voice"
    const val TYPE_LOCATION = "im.location"
    const val TYPE_FILE = "im.file"
    const val TYPE_HTML = "im.html"
    const val TYPE_AD = "im.ad"
    const val TYPE_SYSTEM = "im.system"
    const val TYPE_CUSTOM = "im.custom.*"

    @JvmStatic
    fun getTypeByImMessage(imMessage: ImMessage): String {
        return TYPE_SYSTEM
    }
}